slides/lesson2/solution/main.tf

```diff
@@ -20,38 +20,60 @@ resource "aws_default_vpc" "default" {
   }
 }
 
-
 #===========================================
 #  Elastic Load Balancer
 #===========================================
 resource "aws_elb" "elb" {
-  name                        = "${var.elb_name}-elb-${var.env}"
-  subnets                     = ["${data.aws_subnet_ids.vpc_subnets.ids}"]
-  security_groups             = ["${aws_security_group.web.id}"]
-  tags                        = "${merge(var.tags, map("Name", format("%s", var.elb_name)))}"
-  listener                    = ["${var.elb_listener}"]
-  health_check                = ["${var.elb_health_check}"]
-  cross_zone_load_balancing   = "${var.cross_zone_load_balancing}"
-  connection_draining         = "${var.connection_draining}"
-  connection_draining_timeout = "${var.connection_draining_timeout}"
-  internal                    = "${var.internal}"
-
-  # Variables to enabled SSL on your ELB
-  #ssl_certificate_id = "${data.aws_acm_certificate.cert.arn}"
+  name = "${var.elb_name}-elb-${var.env}"
+  subnets         = data.aws_subnet_ids.vpc_subnets.ids
+  security_groups = [aws_security_group.web.id]
+  tags = merge(
+    var.tags,
+    {
+      "Name" = format("%s", var.elb_name)
+    },
+  )
+  dynamic "listener" {
+    for_each = [var.elb_listener]
+    content {
+      instance_port      = lookup(listener.value, "instance_port", null)
+      instance_protocol  = lookup(listener.value, "instance_protocol", null)
+      lb_port            = lookup(listener.value, "lb_port", null)
+      lb_protocol        = lookup(listener.value, "lb_protocol", null)
+    }
+  }
+  dynamic "health_check" {
+    for_each = [var.elb_health_check]
+    content {
+      healthy_threshold   = lookup(health_check.value, "healthy_threshold", null)
+      interval            = lookup(health_check.value,"interval", null)
+      target              = lookup(health_check.value,"target", null)
+      timeout             = lookup(health_check.value,"timeout", null)
+      unhealthy_threshold = lookup(health_check.value,"unhealthy_threshold", null)
+    }
+  }
+  cross_zone_load_balancing   = var.cross_zone_load_balancing
+  connection_draining         = var.connection_draining
+  connection_draining_timeout = var.connection_draining_timeout
+  internal                    = var.internal
 }
 
 #===========================================
 #  Elastic Load Balancer
 #===========================================
 
+resource "aws_route53_zone" "tf-workshop" {
+  name = var.domain
+}
+
 resource "aws_route53_record" "dns_web" {
```

Comment:
As above, let's keep the DNS-related resources commented out

Reply:
we can't this is required for the intention of the course, this will be managed as an internal DNS